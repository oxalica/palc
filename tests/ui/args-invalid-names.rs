#[derive(palc::Args)]
struct Cli {
    #[arg(long = "")]
    long_empty: i32,
    #[arg(long = "\0")]
    long_control: i32,
    #[arg(long = "-foo")]
    long_dash: i32,
    #[arg(long = "a=b")]
    long_eq: i32,

    #[arg(short = '-')]
    short_dash: i32,
    #[arg(short = '坏')]
    short_unicode: i32,
    #[arg(short)]
    首字符Unicode: i32,
}

fn main() {}
