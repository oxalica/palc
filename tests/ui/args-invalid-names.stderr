error: arg(long) name must NOT be empty
 --> tests/ui/args-invalid-names.rs:3:18
  |
3 |     #[arg(long = "")]
  |                  ^^

error: arg(long) name must NOT contain '=' or ASCII control characters
 --> tests/ui/args-invalid-names.rs:5:18
  |
5 |     #[arg(long = "\0")]
  |                  ^^^^

error: arg(long) name is automatically prefixed by "--", and you should not add more "-" prefix
 --> tests/ui/args-invalid-names.rs:7:18
  |
7 |     #[arg(long = "-foo")]
  |                  ^^^^^^

error: arg(long) name must NOT contain '=' or ASCII control characters
 --> tests/ui/args-invalid-names.rs:9:18
  |
9 |     #[arg(long = "a=b")]
  |                  ^^^^^

error: arg(short) name must NOT be '-' or ASCII control characters
  --> tests/ui/args-invalid-names.rs:12:19
   |
12 |     #[arg(short = '-')]
   |                   ^^^

error: Non-ASCII arg(short) name is reserved. Use `arg(long)` instead. A unicode codepoint is not necessarity a "character" in human sense, thus automatic splitting or argument de-bundling may give unexpected results. If you do want this to be supported, convince us by opening an issue.
  --> tests/ui/args-invalid-names.rs:14:19
   |
14 |     #[arg(short = '坏')]
   |                   ^^^^

error: Non-ASCII arg(short) name is reserved. Use `arg(long)` instead. A unicode codepoint is not necessarity a "character" in human sense, thus automatic splitting or argument de-bundling may give unexpected results. If you do want this to be supported, convince us by opening an issue.
  --> tests/ui/args-invalid-names.rs:17:5
   |
17 |     首字符Unicode: i32,
   |     ^^^^^^^^^^^^^
